sdv/constraints/tabular.py

```diff
@@ -964,21 +961,30 @@ def _reverse_transform(self, table_data):
             pandas.DataFrame:
                 Transformed data.
         """
-        low = table_data[self.low_column_name]
-        high = table_data[self.high_column_name]
-        data = table_data[self._transformed_column]
-
-        data = sigmoid(data, low, high)
-        data = data.clip(low, high)
+        low_diff_column = np.exp(table_data[self.low_diff_column_name]) - 1
+        high_diff_column = np.exp(table_data[self.high_diff_column_name]) - 1
+        if self._dtype != np.dtype('float'):
+            low_diff_column = low_diff_column.round()
+            high_diff_column = high_diff_column.round()
 
         if self._is_datetime:
-            table_data[self.middle_column_name] = pd.to_datetime(data)
-        else:
-            table_data[self.middle_column_name] = data.astype(self._dtype)
+            low_diff_column = convert_to_timedelta(low_diff_column)
+            high_diff_column = convert_to_timedelta(high_diff_column)
 
-        table_data = table_data.drop(self._transformed_column, axis=1)
+        low = table_data[self.low_column_name].to_numpy()
+        if self._is_datetime and self._dtype == 'O':
+            low = cast_to_datetime64(low)
 
-        return table_data
+        middle = pd.Series(low_diff_column + low).astype(self._dtype)
+        table_data[self.middle_column_name] = middle
+        table_data[self.high_column_name] = pd.Series(
+            high_diff_column + middle.to_numpy()
+        ).astype(self._dtype)
+
+        if self.nan_column_name and self.nan_column_name in table_data.columns:
```

Comment:
is it possible for the nan_column_name to exist but not be in the table_data?

Reply:
No, you're right, I changed the condition to check if the nan column is in the table data. I did this first because I thought that the data in `_transform` and` _reverse_transform` could be different, but as we discussed yesterday, it's always the same data that is fitted, transformed, and reverse-transformed in a constraint. Done in [6b77a4d](https://github.com/sdv-dev/SDV/pull/1404/commits/6b77a4d4e7ef55bf738efc7c1e38dd9f5a7f0951)